rfcs/2022014-layernorm-v2/README.md

```diff
@@ -0,0 +1,176 @@
+# RFC: LayerNormalization v2
+
+## Introduction
+
+Several primitives in the library were designed with operation descriptors
+containing only `data_desc` member, which covers source and destination memory
+descriptors. There was an assumption that such operations do not change memory
+format layout through execution nor require a different output data type. And
+never would. Those primitives are batch and layer normalizations, eltwise,
+prelu, lrn, and shuffle.
+
+These operations, except shuffle, assume that only floating-point types will be
+used on input and output of same data type will be produced. It is quite
+essential due to the nature and service of these operations, especially,
+normalizations. Also, outputs of these operations are clearly distinguished only
+in floating-point range of values.
+
+Unfortunately, that assumption did not consider int8 flows when computations may
+contain mixture of quantized data and floating-point data. Mentioned primitives
+are still executed in floating-point data types such as `f32`, `bf16` or `f16`,
+but it usually happens that the following primitives would use quantized output
+from mentioned primitives. It means that every time such situation happens, a
+reorder from floating-point type to integer type appears.
+
+In Transformer-like models, including BERT, the layer normalizations appear in
+every Encoder block. The number of blocks may vary but consider for BERT a base
+case of 12 Encoders resulting in 12 reorders for int8 quantization.
+
+Reorder can easily be incorporated into the layer normalization operator, or any
+other operation, by letting it store the answer in requested data type. The
+problem that `data_desc` does not allow to do that and the only possible
+solution is to extend layer normalization descriptor to accept additional memory
+descriptor. Quantized output of mentioned primitives would not be meaningful in
+terms of operation essence, but this is what's used in production for inference
+and was tested accordingly from accuracy point of view.
+
+### Motivation
+
+Major motivation is free performance improvements for Transformer-like models
+for int8 inference.
+
+## Proposal
+
+### Option 1
```

Comment:
I tried to find an option 2 :)

Reply:
:) Thought this might help in case someone proposed another option, but if there are none I'll certainly change it.